main.rb

```diff
@@ -0,0 +1,179 @@
+module Enumerable
+  def my_each
+    i = 0
+    length.times do
+      yield (self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    length.times do
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    if block_given?
+      arr_ret = []
+      i = 0
+      length.times do
+        arr_ret.push(self[i]) if yield(self[i])
+        i += 1
+      end
+      arr_ret
+    end
+  end
+
+  def my_all?
+    if block_given?
+      # arr_ret=[]
+      retValue = true
+      i = 0
+      length.times do
+        if yield(self[i])
+          retValue = true
+        else
+          return false
+        end
+        i += 1
+      end
+      retValue
+    else
+      i = 0
+      # puts 'no block here'
+      retValue = true
+      length.times do
+        return false unless self[i]
+
+        i += 1
+      end
+      retValue
+    end
+  end
+
+  def my_any?
+    if block_given?
+      retValue = true
+      i = 0
+      length.times do
+        return true if yield(self[i])
+
+        retValue = false unless yield(self[i])
+        i += 1
+      end
+      retValue
+    else
+      i = 0
+      length.times do
+        return true if self[i]
+
+        retValue = false
+        i += 1
+      end
+    end
+    retValue
+  end
```

Comment:
should have working `my_any?` **(covered by automated tests)** identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))